networkx/algorithms/causal/m_separation.py

```diff
@@ -52,27 +64,58 @@ def m_separated(
         for edge_type in [bidirected_edge_name, directed_edge_name]
     ):
         raise RuntimeError(
-            f"m-separation only works on nx with directed and bidirected edges. "
+            f"m-separation only works on graphs with directed and bidirected edges. "
             f"Your graph passed in has the following edge types: {G.edge_types}, whereas "
             f"the function is expecting directed edges named {directed_edge_name} and "
             f"bidirected edges named {bidirected_edge_name}."
         )
+    union_xyz = x.union(y).union(z)
```

Comment:
Similar to `d_separated`, would it make sense to check the acyclic condition of the directed edges?

E.g.
```
    if not nx.is_directed_acyclic_graph(G.get_graphs(directed_edge_name)):
        raise nx.NetworkXError("directed edge graph should be directed acyclic")
```

Reply:
Above this line that is. if the DAG check makes sense to run.

Reply:
Yeah that makes sense, I can add that